relay-server/src/actors/processor.rs

```diff
@@ -2504,6 +2524,110 @@ impl EnvelopeProcessorService {
+            .project_state
+            .has_feature(Feature::StandaloneSpanIngestion);
+        state.managed_envelope.retain_items(|item| match item.ty() {
+            ItemType::OtelSpan | ItemType::EventSpan => {
+                if !standalone_span_ingestion_enabled {
+                    relay_log::warn!("dropping span because feature is disabled");
+                    ItemAction::DropSilently
+                } else {
+                    ItemAction::Keep
+                }
+            }
+            _ => ItemAction::Keep,
+        });
+    }
+
+    fn extract_span_metrics(&self, state: &mut ProcessEnvelopeState) {
+        if state
+            .project_state
+            .has_feature(Feature::StandaloneSpanIngestion)
+        {
+            return;
+        }
+        let span_metrics_extraction_config = match state.project_state.config.metric_extraction {
+            ErrorBoundary::Ok(ref config) if config.is_enabled() => config,
+            _ => return,
+        };
+        let extract_span_metrics = |span: Annotated<Span>| {
+            let span = match self.validate_span(span) {
+                Ok(span) => span,
+                Err(e) => {
+                    relay_log::error!("Invalid span: {e}");
+                    return;
+                }
+            };
+            if let Some(span) = span.value() {
+                crate::metrics_extraction::generic::extract_metrics(
+                    span,
+                    span_metrics_extraction_config,
+                );
+            }
+        };
+        for item in state.managed_envelope.envelope().items() {
+            if item.ty() == &ItemType::OtelSpan {
+                if let Ok(otel_span) =
+                    serde_json::from_slice::<relay_spans::OtelSpan>(&item.payload())
+                {
+                    extract_span_metrics(Annotated::new(otel_span.into()));
+                }
+            } else if item.ty() == &ItemType::EventSpan {
+                if let Ok(event_span) = Annotated::<Span>::from_json_bytes(&item.payload()) {
+                    extract_span_metrics(event_span);
+                }
+            }
+        }
+    }
+
+    #[cfg(feature = "processing")]
+    fn process_spans(&self, state: &mut ProcessEnvelopeState) {
+        let mut items: Vec<Item> = Vec::new();
+        let mut add_span = |span: Annotated<Span>| {
+            let span = match self.validate_span(span) {
+                Ok(span) => span,
+                Err(e) => {
+                    relay_log::error!("Invalid span: {e}");
+                    return;
+                }
+            };
+            if let Ok(payload) = span.to_json() {
+                let mut item = Item::new(ItemType::Span);
+                item.set_payload(ContentType::Json, payload);
+                items.push(item);
+            }
+        };
+        let standalone_span_ingestion_enabled = state
+            .project_state
+            .has_feature(Feature::StandaloneSpanIngestion);
```

Comment:
nit: In theory we don't have to check this feature flag here because it is already done in `filter_spans`.

Reply:
Could someone send spans through an old Relay version (not having this filtering logic) and somehow have spans be ingested?

I added this logic to potentially protect against cases like that.

Reply:
> Could someone send spans through an old Relay version (not having this filtering logic) and somehow have spans be ingested?

Those would still get caught by the `filter_spans` function of the newer Relay, so I think it's safe to remove here.